tests/testthat/test-corp_text.R

```diff
@@ -30,8 +30,75 @@ test_that("corp_text_rbindlist", {
+        expect_equal(sep$tokens[1]$type, "NA")
+        expect_equal(sep$tokens[4]$type, "NA")
+
+        # Now append a corp_text object for the paragraph, and our new sep object
+        corp_list <- rlist::list.append(corp_list, CorporaCoCo::corp_text(para, tokens = NULL), sep)
+    }
+    expect_length(corp_list, 10)  # 5 paras each followed by a sep (boundary marker)
+    expect_true(is.corp_text(corp_list[[1]]))
+    expect_true(is.data.table(corp_list[[1]]$tokens))
+    expect_equal(corp_list[[1]]$tokens[2][, token], "Sherlock")
+    expect_equal(corp_list[[9]]$token[10][, token], "coffee")
+    expect_equal(corp_list[[9]]$token[11][, token], "pot")
+
+    combined <- corp_text_rbindlist(corp_list)
+    expect_true(is.corp_text(combined))
+
+    expect_equal(corp_get_text(combined), "Mr. Sherlock Holmes, who was usually very late in the mornings, save upon those not infrequent occasions when he was up all night, was seated at the breakfast table. I stood upon the hearth-rug and picked up the stick which our visitor had left behind him the night before. It was a fine, thick piece of wood, bulbous-headed, of the sort which is known as a \"Penang lawyer.\" Just under the head was a broad silver band nearly an inch across. \"To James Mortimer, M.R.C.S., from his friends of the C.C.H.,\" was engraved upon it, with the date \"1884.\" It was just such a stick as the old-fashioned family practitioner used to carry--dignified, solid, and reassuring.  _b_ _b_ _b_ _b_ _b_  \"Well, Watson, what do you make of it?\"  _b_ _b_ _b_ _b_ _b_  Holmes was sitting with his back to me, and I had given him no sign of my occupation.  _b_ _b_ _b_ _b_ _b_  \"How did you know what I was doing? I believe you have eyes in the back of your head.\"  _b_ _b_ _b_ _b_ _b_  \"I have, at least, a well-polished, silver-plated coffee-pot in front of me,\" said he. \"But, tell me, Watson, what do you make of our visitor's stick? Since we have been so unfortunate as to miss him and have no notion of his errand, this accidental souvenir becomes of importance. Let me hear you reconstruct the man by an examination of it.\"  _b_ _b_ _b_ _b_ _b_ ")
+    tokens <- corp_get_tokens(combined)
+
+    # Check the real text
+    #print(tokens, as_data_table=TRUE)
+    expect_equal(tokens[token == "Penang", "start"]$start, 360)
+    expect_equal(tokens[type == "penang", "start"]$start, 360)
+    expect_equal(tokens[token == "Penang", "end"]$end, 365)
+    expect_equal(tokens[type == "penang", "end"]$end, 365)
+
+    # This token appears twice...
+    expect_equal(tokens[token == "Holmes", "start"]$start, c(14, 749))
+
+    # This token appears four times...
+    expect_equal(tokens[token == "and", "start"]$start, c(195, 649, 789, 1166))
+
+    # Now check the boundary markers
+    expect_equal(tokens[token == "occupation", "end"]$end, 832)  #  immediately before a boundary
+    expect_equal(tokens[token == "Well", "start"]$start, 688)  #  immediately after a boundary
```

Comment:
very minor, but 2 white spaces after the hashtag for the comment, rather than 1. (expected 2 before, but 1 after)

Reply:
Done